adapta/storage/models/aws.py

```diff
@@ -62,6 +58,18 @@ def from_uri(cls, url: str) -> "S3Path":
     path: str
     protocol: str = DataProtocols.S3.value
 
+    def __post_init__(self):
+        if not self.bucket:
+            raise ValueError("Bucket must be defined")
+
+        path_regex = r"^(?![0-9]+\.[0-9]+\.[0-9]+\.[0-9]+$)[a-z0-9]([a-z0-9\-]{1,61}[a-z0-9])?(\/(?!.*(\/\/|\\))([^\/].{0,1022}\/?)?)?$"
+
+        s3_path_without_prefix = f"{self.bucket}/{self.path}"
```

Comment:
regex should only be applied to `self.path`, bucket is checked on line 62, we do not aim to validate bucket name with this PR, out of scope